### PR TITLE
[AppProvider & ThemeProvider] remove ndl

### PR DESCRIPTION
**Pull request by alex-page**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-ux/issues/531

### WHAT is this pull request doing?

- [x] Makes the `theme` property optional, light color scheme is the default
- [x] Removes legacy topbar theming
- [x] Removes legacy theming functions
- [ ] Fixes ThemeProvider tests

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
- [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

**Comment by BPScott** [comment]
i'm probably gonna be pretty full for the rest of today. I'll get to this on Monday :)

File: src/components/AppProvider/AppProvider.tsx [hunk 1]
```diff
@@ -89,7 +89,7 @@ export class AppProvider extends Component<AppProviderProps, State> {
     const {theme = {}, children} = this.props;
 
     const {intl, link} = this.state;
-    const features = {newDesignLanguage: false, ...this.props.features};
+    const features = {...this.props.features};
```
Comment (BPScott) [inline src/components/AppProvider/AppProvider.tsx:92]: No need to spread and create a new object anymore. We can get rid of this variable and do the following below

```
<FeaturesContext.Provider value={this.props.features}>
```

File: src/components/Filters/Filters.tsx [hunk 2]
```diff
@@ -390,6 +390,7 @@ class FiltersInner extends Component<CombinedProps, State> {
 
     const filtersContainerMarkup = isNavigationCollapsed ? (
       <Sheet
+        accessibilityLabel={moreFiltersLabel}
```
Comment (BPScott) [inline src/components/Filters/Filters.tsx:393]: This Sheet stuff seems unrelated to this PR? Pull it out to a separate PR?

File: src/components/PolarisTestProvider/PolarisTestProvider.tsx [hunk 3]
```diff
@@ -77,14 +77,12 @@ export function PolarisTestProvider({
 
   const uniqueIdFactory = new UniqueIdFactory(globalIdGeneratorFactory);
 
-  const features = {newDesignLanguage: false, ...featuresProp};
+  const features = {...featuresProp};
```
Comment (BPScott) [inline src/components/PolarisTestProvider/PolarisTestProvider.tsx:80]: No need to spread anymore and create a new object anymore. We can remove this declaration and avoid renaming the features prop above

```diff
- features: featuresProp = {},
+ features = {}
```

File: src/components/ThemeProvider/ThemeProvider.tsx [hunk 4]
```diff
@@ -26,7 +25,7 @@ interface ThemeProviderThemeConfig extends Discard<ThemeConfig, 'colorScheme'> {
 
 interface ThemeProviderProps {
   /** Custom logos and colors provided to select components */
-  theme: ThemeProviderThemeConfig;
+  theme?: ThemeProviderThemeConfig;
```
Comment (BPScott) [inline src/components/ThemeProvider/ThemeProvider.tsx:28]: When would we not passing a theme into a theme provider make sense?

File: src/components/ThemeProvider/tests/ThemeProvider.test.tsx [hunk 5]
```diff
@@ -1,31 +1,14 @@
 import React, {useContext} from 'react';
-import {createMount} from 'test-utilities';
+import {mountWithApp} from 'test-utilities/react-testing';
```
Comment (BPScott) [inline src/components/ThemeProvider/tests/ThemeProvider.test.tsx:2]: The `test-utilities` index reexports everything from `test-utilities/react-testing` for the shortest naming

```suggestion
import {mountWithApp} from 'test-utilities';
```

File: src/utilities/theme/utils.ts [hunk 6]
```diff
@@ -23,34 +22,28 @@ interface CustomPropertiesConfig extends ThemeConfig {
...
+  });
 }
 
 export function buildThemeContext(
   themeConfig: ThemeConfig,
   cssCustomProperties?: CustomPropertiesLike,
 ): Theme {
   const {logo, colors = {}, colorScheme} = themeConfig;
-  const {topBar, ...newDesignLanguageColors} = colors;
+  const {...colorValues} = colors;
```
Comment (BPScott) [inline src/utilities/theme/utils.ts:42]: No need to spread this to create a new value. Below we can use colors directly


```diff
- colors: colorValues,
+ colors
```

**Review by BPScott — COMMENTED** [review]
Quick first pass

File: src/components/ThemeProvider/ThemeProvider.tsx (hunk 4 above)
Comment (alex-page) [reply src/components/ThemeProvider/ThemeProvider.tsx:28]: If nothing is passed in then it returns the default CSS variables. What do you think about that assumption?

Comment (BPScott) [reply src/components/ThemeProvider/ThemeProvider.tsx:28]: I would have expected a theme without an config should be a non-operation:

```
<ThemeProvider theme={{colorScheme: 'dark'}}>
  <ThemeProvider>
    <Button>I would expect this button to still be in a dark scheme</Button>
  </ThemeProvider>
</ThemeProvider>
```

Forcing a color scheme of light seems to breakt that

**Comment by BPScott** [comment]
Dont' forget removing mentions of new DL in the ThemeProvider/AppProvider readmes

**Comment by BPScott** [comment]
I've updated low-hanging fruit on the ThemeProvider and AppProvider readmes. 


What do you think we should do about the AppProvider's "with theme" and "with theme using all keys" examples? They currently document the legacy theming of the topbar. Should we remove them and instead point people at the ThemeProvider documentation or should we rework them to describe modern theming?

**Comment by alex-page** [comment]
> What do you think we should do about the AppProvider's "with theme" and "with theme using all keys" examples? They currently document the legacy theming of the topbar. Should we remove them and instead point people at the ThemeProvider documentation or should we rework them to describe modern theming?

I think we should be removing the ability to theme the `TopBar` with the `theme`. You can see the new way to theme `TopBar` in `Top bar themed for Plus`. The only useful information would be the `logo`, `colorScheme` or `color` overrides.

**Comment by kyledurand** [comment]
> I would hope to deprecate the theming of the TopBar using the previous API.

I agree but that would break the current contextual save bar in web since it uses the logo and not contextControl. Not a deal breaker just something we'd have to update in web before this ships

File: src/components/Popover/components/PopoverOverlay/PopoverOverlay.tsx [hunk 7]
```diff
@@ -51,7 +51,7 @@ export interface PopoverOverlayProps {
   fixed?: boolean;
   hideOnPrint?: boolean;
   onClose(source: PopoverCloseSource): void;
-  colorScheme?: InversableColorScheme;
+  colorScheme?: NonNullable<ThemeProviderProps['theme']>['colorScheme'];
```
Comment (kyledurand) [inline src/components/Popover/components/PopoverOverlay/PopoverOverlay.tsx:54]: Curious what `NonNullable` is doing here if the prop is optional?

File: src/components/ThemeProvider/ThemeProvider.tsx [hunk 8]
```diff
@@ -121,19 +93,13 @@ export function ThemeProvider({
...
-  colorScheme: InversableColorScheme | undefined,
-  parentColorScheme: OriginalColorScheme | undefined,
-) {
+  colorScheme: ThemeConfig['colorScheme'],
+  parentColorScheme?: Theme['colorScheme'],
+): Theme['colorScheme'] {
   if (colorScheme == null) {
     return parentColorScheme || 'light';
-  } else if (isInverseColorScheme(colorScheme)) {
+  } else if (colorScheme === 'inverse') {
```
Comment (kyledurand) [inline src/components/ThemeProvider/ThemeProvider.tsx:102]: ✨  Nice and simple, I'm typically a fan of typeguards but this makes more sense in this case

**Review by kyledurand — APPROVED** [review]
Lots changes but the code and readmes look good.

File: src/components/ThemeProvider/ThemeProvider.tsx (hunk 8 above)
Comment (BPScott) [reply src/components/ThemeProvider/ThemeProvider.tsx:102]: This is still a typeguard, it's just one that TS can work out automatically instead of us having  to be explicit.

Inside this if block the type of `colorScheme` is still narrowed down `'light'| 'dark' | 'inverse'` to `'light' | 'dark'`  (`undefined` was already narrowed away thanks to the prior if block)

File: src/components/Popover/components/PopoverOverlay/PopoverOverlay.tsx (hunk 7 above)
Comment (BPScott) [reply src/components/Popover/components/PopoverOverlay/PopoverOverlay.tsx:54]: Note where the Generic ends (spacing added for emphasis) 

```
NonNullable<ThemeProviderProps['theme']>    ['colorScheme']
```

`ThemeProviderProps['theme']` is  `ThemeConfig | undefined`. The NonNullable knocks off the undefined bit of that so we have an object and thus can access the type of `ThemeConfig['colorScheme']` (which is itself potentially undefined)

**Comment by BPScott** [comment]
I would hope to deprecate the theming of the TopBar using the previous API.

That's what this PR does. Here's where you rip out being able to define `{colors: {topBar: {background: '#225062'}}` in the theme config: https://github.com/Shopify/polaris-react/pull/3849/files#diff-d20bc9ba53e6c4c404e50adf23cb39bf5fe6525effc3acd98867dacd454826b0L84


This is the configuration that the ["With Theme"](https://github.com/Shopify/polaris-react/blob/1c32f482c955e42b27b2b214af53062d46ad853d/src/components/AppProvider/README.md) example demonstrates

**Comment by alex-page** [comment]
I think we should remove the logo and other parts in the theme provider that are no longer used. We should fix the contextual save bar in web to resolve this issue https://github.com/Shopify/web/issues/32812

File: src/utilities/features/types.ts [hunk 9]
```diff
@@ -3,6 +3,5 @@ export interface FeaturesConfig {
 }
 
 export interface Features {
-  newDesignLanguage: boolean;
   [key: string]: boolean;
```
Comment (BPScott) [inline src/utilities/features/types.ts:6]: TODO: Remove this too so that people who use the newDesignLanguage prop will be alerted that it has gone away.

Features should be an empty object: `export interface Features {}`

Which isn't very useful now but let's keep it around in case we want to add toggles in the future

**Review by BPScott — COMMENTED** [review]
see inline TODO comment

**Review by BPScott — APPROVED** [review]
Pending final percy run 🎉
